src/registrar/assets/js/get-gov-admin.js

```diff
@@ -750,3 +760,107 @@ function initializeWidgetOnList(list, parentId) {
+            });
+        }
+    });
+
+    function handleFederalAgencyChange(federalAgency, organizationType) {
+        // Set the org type to federal if an agency is selected
+        let selectedText = federalAgency.find("option:selected").text();
+        if (selectedText !== "Non-Federal Agency" && selectedText) {
+            if (organizationType.value !== "federal") {
+                organizationType.value = "federal";
+            }
+            // Set the SO field
+        }else if (selectedText === "Non-Federal Agency" && organizationType.value === "federal") {
+            organizationType.value = "";
+            // Set the SO field
+        }
+        
+        // There isn't a senior official associated with null records and non federal agencies
+        if (!selectedText || selectedText === "Non-Federal Agency") {
+            return;
+        }
+
+        // Get the associated senior official with this federal agency
+        let $seniorOfficial = django.jQuery("#id_senior_official");
+        if (!$seniorOfficial) {
+            console.log("Could not find the senior official field");
+            return;
+        }
+
+        let seniorOfficialApi = document.querySelector("#senior_official_from_agency_json_url").value;
+        fetch(`${seniorOfficialApi}?agency_name=${selectedText}`)
+        .then(response => response.json())
+        .then(data => {
+            if (data.error) {
+                console.error('Error in AJAX call: ' + data.error);
+                return;
+            }
+
+            let seniorOfficialId = data.id;
+            let seniorOfficialName = [data.first_name, data.last_name].join(" ");
+            if (!seniorOfficialId || !seniorOfficialName || !seniorOfficialName.trim()){
+                console.error("Could not assign current Senior Official: no values found.")
+                return;
+            }
+
+            // Add the senior official to the dropdown.
+            // This format supports select2 - if we decide to convert this field in the future.
+            if ($seniorOfficial.find(`option[value='${seniorOfficialId}']`).length) {
+                // Select the value that is associated with the current Senior Official.
+                $seniorOfficial.val(seniorOfficialId).trigger("change");
+            } else { 
+                // Create a DOM Option that matches the desired Senior Official. Then append it and select it.
+                let userOption = new Option(seniorOfficialName, seniorOfficialId, true, true);
+                $seniorOfficial.append(userOption).trigger("change");
+            }
+        })
+        .catch(error => console.error('Error fetching senior official:', error));
+    }
+
+    function handleStateTerritoryChange(stateTerritory, urbanizationField) {
+        let selectedValue = stateTerritory.value;
+        if (selectedValue === "PR") {
+            showElement(urbanizationField)
+        } else {
+            hideElement(urbanizationField)
```

Comment:
This is great work! On possible nit-pick: do we want to clear this on change? If the urbanization field is only hidden but not cleared, could that result in an urbanization value being saved for an org that shouldn't have one? I'm not totally clear on how that works.

Reply:
I would clear on save, not on the change. That way if you accidentally were messing a portfolio and changed that field you don't accidentally erase data without clicking save.